packages/drivers/odsp-driver/src/fetchSnapshot.ts

```diff
@@ -101,18 +96,13 @@ export async function fetchSnapshot(
 	}
 
 	const queryString = getQueryString(queryParams);
-	const { url, headers } = getUrlAndHeadersWithAuth(
-		`${snapshotUrl}${path}${queryString}`,
-		token,
-		forceAccessTokenViaAuthorizationHeader,
-	);
+	const url = `${snapshotUrl}${path}${queryString}`;
 	const response = (await PerformanceEvent.timedExecAsync(
 		logger,
 		{
 			eventName: "fetchSnapshot",
-			headers: Object.keys(headers).length > 0 ? true : undefined,
 		},
-		async () => snapshotDownloader(url, { headers }),
+		async () => snapshotDownloader(url),
```

Comment:
Where do we fetch token in this flow? Did it move to snapshotDownloader()?
If so, is the expectation that token is fetched inside of snapshotDownloader(), or that it's done outside (and captured when creating snapshotDownloader)? This might impact where getWithRetryForTokenRefresh() loop is and its effectiveness (of getting refreshed token into this flow).

I think we need to capture somehow expectations / requirements in comments

Reply:
Token fetching is moved to snapshotDownloader. Luckily, snapshotDownloader definition was already inside getWithRetryForTokenRefresh, so it was straight forward to move it there. I will add comments where appropriate.